### PR TITLE
two design patterns

### PR DESCRIPTION
**Pull request by zaynba**
uploaded markdown file describing two design patterns

### CODE REVIEW

File: Requirements/DesignPatterns.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      - Handles user account creation, modification, and profile updates.
+      - limits user access based on their user type and login status
+
+    - b.) Event Model:
+
+      - Manages events, tickets, and related statistics.
+      - Handles the creation, modification, and review of events by sellers.
+
+    - c.) Transaction Model:
+      - Manages transactions, including ticket purchases and refunds.
```
Comment (tatkg24) [inline Requirements/DesignPatterns.md:45]: We are not including refunds in our model at this point in time so probably remove this. Instead, you could be more specific about the buyer and seller transaction mentioning that the transaction model will handle the funds transitioning from the buyer's specified payment option and into the seller's account.

**Review by tatkg24 — CHANGES_REQUESTED** [review]
### **In regards to the section about MVC:**

1. We are not including refunds in our model at this point in time so probably remove this from line 45 (the section about transactions). Instead, you could be more specific about the buyer and seller transaction mentioning that the transaction model will handle the funds transitioning from the buyer's specified payment option and into the seller's account.

2. In part 2, where you discuss the view, I would include a bullet point about the guest user, specifically how the sign-up page allows them to join the platform as a buyer or seller.

### **In regards to the section about the observer pattern:**

1. In the description (line 75), instead of writing "for a ticketing platform", reword this to say "Another suitable design pattern for our platformTickthub is XYZ". This way we are being more specific since this is our project and these are patterns for us to use; not some arbitrary system. 

2. In the point about **observers** I don't believe we have integrated alerts into the system for users. Perhaps, you could remove the language about notifications and instead say something about how the interface allows each user to monitor their user-specific activity?

Otherwise, everything else looks great Zaynb! Let me know if you need further clarification or would like me to review again :)

**Review by JoheeYeom0103 — COMMENTED** [review]
The choice of design patterns is well considered of application device and performance optimisation and is well backed up by specific features!

The only suggestions would be from line 95~98, we could  
specify how observer pattern could benefit administrator as utilizing this pattern would play a significant role for them to keep track of user history/activities. In particular, we could highlight the interaction between the administrator and seller in the process of making post request and granting permission process.
